File: project_euler/02/028.py
#!/usr/bin/env python3
'''Copyright [2017] <dmnsn7@gmail.com>'''

N = 1001


def main():
    vec = [1, 24, 52, 32]
    for _ in range(N // 2):
        vec = [vec[0] + vec[1], vec[1] + vec[2], vec[2] + vec[3], vec[3]]

    print(vec[0])


if __name__ == '__main__':
    main()

File: project_euler/02/028.go
// Copyright [2017] <dmnsn7@gmail.com>

package main

import "fmt"

const cN = 1001

func main() {
	v := []int{1, 24, 52, 32}
	for i := 0; i < cN/2; i++ {
		v = []int{v[0] + v[1], v[1] + v[2], v[2] + v[3], v[3]}
	}

	fmt.Printf("%d\n", v[0])
}
